arch/arc/core/cache.c

```diff
@@ -67,24 +67,7 @@ static void dcache_enable(void)
 	dcache_dc_ctrl(DC_CTRL_DC_ENABLE);
 }
 
-
-/**
- *
- * @brief Flush multiple d-cache lines to memory
- *
- * No alignment is required for either <start_addr> or <size>, but since
- * dcache_flush_mlines() iterates on the d-cache lines, a cache line
- * alignment for both is optimal.
- *
- * The d-cache line size is specified either via the CONFIG_CACHE_LINE_SIZE
- * kconfig option or it is detected at runtime.
- *
- * @param start_addr the pointer to start the multi-line flush
- * @param size the number of bytes that are to be flushed
- *
- * @return N/A
- */
-static void dcache_flush_mlines(u32_t start_addr, u32_t size)
+void arch_cache_flush(void *start_addr, size_t size)
 {
 	u32_t end_addr;
```

Comment:
Should use `uintptr_t`.

p.s. I understand that we only support 32-bit ARC at the moment, but using `uintptr_t` here would ensure that no code change is necessary when 64-bit ARC support is ever added.

Reply:
Agreed. Changed it to uintptr_t type.